src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzerTests.cs

```diff
@@ -131,6 +131,54 @@ class Target
             await VerifyAnalyzerAsyncCs(source, tfmAndOption);
         }
 
+        [Fact]
+        public async Task MethodsWithOsDependentTypeParameterWarns()
+        {
+            var csSource = @"
+using System.Runtime.Versioning;
+
+[SupportedOSPlatform(""windows"")]
+class WindowsOnlyType {}
+
+public class Test
+{
+    void GenericMethod<T>() {}
+    void GenericMethod2<T1, T2>() {}
+    void M1()
+    {
+        [|GenericMethod<WindowsOnlyType>()|];
```

Comment:
What is the diagnostic message here? Will the message indicate that the core issue is with the type argument?

Reply:
The message will be `'WindowsOnlyType' is supported on 'windows'`, because ITypeArgument returns just the type, but i think it is enough, what you think?

Reply:
That sounds reasonable. Might be confusing if the type argument was implicitly figured out by the compiler, but I think it is okay.